partner_category_unique_per_organization/models/partner.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2023 Akretion (https://www.akretion.com).
+# @author Pierrick Brun <pierrick.brun@akretion.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, models
+from odoo.exceptions import UserError
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    @api.constrains("category_id")
+    def _check_category_unique_per_organization(self):
+        for record in self:
+            for categ in record.category_id.filtered(
+                lambda categ: categ.unique_per_organization
+            ):
```

Comment:
isn't `record.category_id.filtered("unique_per_organization")` enough and more readable than a lambda?